src/Compilers/CSharp/Portable/Binder/PatternVariableFinder.cs

```diff
@@ -165,6 +165,17 @@ public override void VisitArgument(ArgumentSyntax node)
                 return;
             }
 
+            switch (node.Identifier.Parent.Parent.Parent.Kind())
```

Comment:
Added comments for each parent.

According to Language Specification, out/ref arguments are not allowed in the list of arguments for element access. Out variable declarations are disallowed in this context during the parsing.